src/test/java/com/fasterxml/jackson/databind/ser/CanonicalJsonFactory.java

```diff
@@ -0,0 +1,30 @@
+package com.fasterxml.jackson.databind.ser;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.math.BigDecimal;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.io.IOContext;
+import com.fasterxml.jackson.core.JsonFactory;
+
+/**
+ * TODO Fix double numbers. This feels like a very heavy solution plus I can't
+ * use the JsonFactory.builder().
+ */
+public class CanonicalJsonFactory extends JsonFactory {
+    private static final long serialVersionUID = 1L;
+
+    private ValueToString<BigDecimal> _serializer;
+
+    public CanonicalJsonFactory(ValueToString<BigDecimal> serializer) {
+        this._serializer = serializer;
+    }
+
+    @Override
+    protected JsonGenerator _createGenerator(Writer out, IOContext ioCtxt)
```

Comment:
I think that instead of requiring sub-classing, we could add a "delegate wrapper" configuration in 2.16 for `JsonFactory` (via Builder); basically ability to wrap constructed `JsonGenerator` into delegate, similar to how `InputDecorator`/`OutputDecorator` allow wrapping of `InputStream` / `OutputStream`.
This could be generally useful for other uses too, when overriding JsonParser/JsonGenerator functionality.

Reply:
Like in my last commit (related MR in core: https://github.com/FasterXML/jackson-core/pull/1051)?